File: Base/Workspace/Domain/Base/Workspace/Data/PullResponse.cs
namespace Allors.Workspace.Data
{
    using System.Collections.Generic;

    public class PullResponse
    {
        public string userSecurityHash { get; set; }

        public string[][] objects { get; set; }

        public Dictionary<string, string> namedObjects { get; set; }

        public Dictionary<string, string[]> namedCollections { get; set; }

        public Dictionary<string,object> namedValues { get; set; }
    }
}

File: dotnet/core/workspace/typescript.legacy/site/wwwroot/core/workspace/data/responses/pullresponse.ts
namespace Allors.Data {
    export interface PullResponse extends  Response {
        userSecurityHash: string;
        objects?: string[][];

        namedObjects?: { [id: string]: string; };
        namedCollections?: { [id: string]: string[]; };
        namedValues?: { [id: string]: any; };
    }
}